src/main/kotlin/lotto/domain/number/Numbers.kt

```diff
@@ -0,0 +1,34 @@
+package lotto.domain.number
+
+class Numbers private constructor(val winningNumbers: List<Number>) {
+
+    companion object {
+        const val WINNING_NUMBER_COUNT = 6
+
+        fun fromInts(vararg numbers: Int): Numbers {
+            if (numbers.distinct().size != WINNING_NUMBER_COUNT) {
+                "Winning numbers must be $WINNING_NUMBER_COUNT unique values"
+            }
+
+            val numberList = numbers.map { Number(it) }
+            return Numbers(numberList)
+        }
+    }
+
+    init {
+        require(winningNumbers.size == winningNumbers.distinct().size) {
+            throw IllegalArgumentException("Winning numbers must not contain duplicates")
+        }
+        require(winningNumbers.size == WINNING_NUMBER_COUNT) {
+            throw IllegalArgumentException("Winning numbers size must equal to $WINNING_NUMBER_COUNT")
+        }
+    }
+
+    fun match(otherNumbers: Numbers): Int {
+        return winningNumbers.count { it in otherNumbers.winningNumbers }
+    }
+
+    fun match(bonusNumber: BonusNumber): Boolean {
+        return winningNumbers.any { bonusNumber.number == it.number }
+    }
```

Comment:
Having the same function name with different return types don't sound good to me.
How do you think about it?
<br>
(My suggestion)

```suggestion
    fun countMatches(otherNumbers: Numbers): Int {
        return winningNumbers.count { it in otherNumbers.winningNumbers }
    }

    fun hasBonusNumber(bonusNumber: BonusNumber): Boolean {
        return winningNumbers.any { bonusNumber.number == it.number }
    }
```

Reply:
https://github.com/kotlin-study-dh/kotlin-lotto-6/pull/2#discussion_r2158834309
I initially used polymorphism to overload the method, but as you pointed out, having different return types can cause confusion. So I revised the method accordingly. Thanks for the great feedback!